src/order/ideal.lean

```diff
@@ -201,6 +240,18 @@ instance : lattice (ideal P) :=
 
 end semilattice_sup_bot
 
+section semilattice_inf
+
+variable [semilattice_inf P]
+
+/-- A prime ideal is an ideal that satisfies `x ⊓ y ∈ I → x ∈ I ∨ y ∈ I`
+-/
+@[mk_iff] structure is_prime (I : ideal P) : Prop :=
```

Comment:
Wikipedia tells me (at [prime ideals](https://en.wikipedia.org/wiki/Ideal_(order_theory)#Prime_ideals)) that there's a more general definition of prime ideal which works even without the `[semilattice_inf P]` assumption (namely, that the complement is an order filter), and reduces to this definition when the `[semilattice_inf P]` instance exists.

Would it be much more work to have the more general definition, and prove that your definition is an equivalent characterization in the `semilattice_inf` case?

The file `order/pfilter.lean` which defines order filters imports `order/ideal.lean`, so you might have to merge the two files to avoid circular imports. Honestly when I added `order/pfilter.lean` recently I wasn't sure whether it should have been in this file all along, but this would be a good argument for having a single file.

Reply:
I didn't know that!
I'll try and work on it.

Maybe the solution to the import problem is to create another file for properties of ideals, such as prime and maximal?